src/datalog/data_access_analysis.dl

```diff
@@ -343,5 +343,41 @@ propagated_data_access(NextEA,Mult,EA_ref):-
         data_access_pattern(EA_ref,S,Mult,From)
     ).
 
+// last data_access_pattern_candidate
+.decl last_data_access_candidate(EA:address,Where:address)
+.output last_data_access_candidate
+
+last_data_access_candidate(EA,Where):-
+    data_access_pattern_candidate(EA,_,_,_),
+    Where = max Where : {data_access_pattern_candidate(Where,_,_,_), Where < EA}.
+
+// adjacent non collision data accesses
+.decl non_collision_access(EA:address,EA_ref:address)
+.output non_collision_access
+
+non_collision_access(EA,EA):-
+    data_access_pattern_candidate(EA,_,Mult,_),
+    Mult > 0.
+
+// connecting adjacent data accesses when they have the same multiplier
+// and they are not colliding
+non_collision_access(EA_prop,EA_access):-
+    last_data_access_candidate(EA_prop,EA_access),
+    data_access_pattern_candidate(EA_prop,_,Mult,_),
+    data_access_pattern_candidate(EA_access,_,Mult,_), Mult > 0,
+    (EA_prop - EA_access) % as(Mult,unsigned) = 0.
```

Comment:
This rule can match very distant data accesses if they happen to have the same multiplier. E.g. if we have a data access at address `0`  with multiplier 2 and the next one is at address `10000` with the same multiplier, we will get the tuple `non_collision_access(10000,0)`. I don't think that is what we want, right?

At the very least, I think it might make sense to limit the distance to a small number of multipliers, e.g.:
```
EA_prop - EA_access / Mult <= 2
```
I don't know what the right number would be, what cases did you have in mind for this rule @ucasqsl ?